### PR TITLE
Author guide fixups

### PR DESCRIPTION
**Pull request by joe-lawrence**
Two small tweaks for the patch author guide.  The first clarifies the context for load hooks (stop_machine and module notifier).  The second adds a shadow variable - data semantic example.

I left the locking example todo in place -- @jpoimboe -- would `0010-CVE-2014-2706-2.patch` that you mailed me last week work here?  If so I can include that as well.

### CODE REVIEW

File: doc/patch-author-guide.md [hunk 1]
```diff
@@ -113,8 +113,15 @@ patch module is loaded.
 
 `kpatch-macros.h` provides `KPATCH_LOAD_HOOK` and `KPATCH_UNLOAD_HOOK` macros
 to define such functions.  The signature of both hook functions is `void
-foo(void)` and and they may run in `stop_machine` context (so they must not
-sleep).
+foo(void)`.  Their execution context is as follows:
+
+* For objects that are already loaded, hook functions will be run by
```
Comment (jpoimboe) [inline doc/patch-author-guide.md]: The term 'object' has a specific meaning in our code, but some readers of this doc might not grok what it means.  So maybe this should say something like "For patches to vmlinux or for kernel modules which have already been loaded ..."

File: doc/patch-author-guide.md [hunk 2]
```diff
@@ -113,8 +113,15 @@ patch module is loaded.
...
+`stop_machine` as part of applying or removing a patch. (Therefore the hooks
+must not block or sleep.)
+
+* For objects not yet loaded, a module-notifier will execute load hooks when
```
Comment (jpoimboe) [inline doc/patch-author-guide.md]: Similar change here: "For patches to kernel modules which haven't been loaded yet..."

**Comment by joe-lawrence** [comment]
v2
  - incorporated 'object' terminology suggestions
  - added locking semantic text

**Comment by jpoimboe** [comment]
Good stuff!

👍

**Pull request merged**